synapse/handlers/federation.py

```diff
@@ -2027,9 +2028,15 @@ def do_auth(self, origin, event, context, auth_events):
         """
         room_version = yield self.store.get_room_version(event.room_id)
 
-        yield self._update_auth_events_and_context_for_auth(
-            origin, event, context, auth_events
-        )
+        try:
+            yield self._update_auth_events_and_context_for_auth(
+                origin, event, context, auth_events
+            )
+        except Exception:
+            # We don't really mind if the above fails, so lets not fail
+            # processing if it does.
+            logger.exception("Failed to call _update_auth_events_and_context_for_auth")
```

Comment:
should we really log a full stacktrace here? they look incredibly scary in the logs when they happen, so if it's something that we don't count as an error, we shouldn't log the stacktrace. just `logger.warning` with the type and value of the exception.

Reply:
Hmm, maybe. We shouldn't have any expected exceptions (now), so I'm slightly worried about not having the stack trace at all.